mirdata/jams_utils.py

```diff
@@ -0,0 +1,257 @@
+                    + 'but is a list of something else')
+            jam.annotations.append(sections_to_jams(sections))
+
+    # multi-sections (sections with multiple levels)
+    if multi_section_data is not None:
+        if type(multi_section_data) != list:
+            raise TypeError(
+                'multi_section_data should be a list of tuples')
+        for sections in multi_section_data:
+            if type(sections) != tuple:
+                raise TypeError(
+                    'multi_section_data should be a list of tuples, '
+                    + 'but is a list of something else')
+            if (type(sections[0]) != list) or (type(sections[0][0]) != tuple):
+                raise TypeError(
+                    'tuples in multi_section_data should contain a '
+                    + 'list of tuples, indicating annotations in the different '
+                    + 'levels, e.g. ([(segments0, level0), '
+                    + '(segments1, level1)], annotator)'
+                    )
+            jam.annotations.append(multi_sections_to_jams(sections))
+
+    # chords
+    if chord_data is not None:
+        if type(chord_data) != list:
+            raise TypeError(
+                'chord_data should be a list of tuples')
+        for chords in chord_data:
+            if type(chords) != tuple:
+                raise TypeError(
+                    'chord_data should be a list of tuples, '
+                    + 'but is a list of something else')
+            jam.annotations.append(chords_to_jams(chords))
+
+    # keys
+    if key_data is not None:
+        if type(key_data) != list:
+            raise TypeError(
+                'key_data should be a list of tuples')
+        for keys in key_data:
+            if type(keys) != tuple:
+                raise TypeError(
+                    'key_data should be a list of tuples, '
+                    + 'but is a list of something else')
+            jam.annotations.append(keys_to_jams(keys))
+
+    # f0
+    if f0_data is not None:
+        if type(f0_data) != list:
+            raise TypeError(
+                'f0_data should be a list of tuples')
+        for f0s in f0_data:
+            if type(f0s) != tuple:
+                raise TypeError(
+                    'f0_data should be a list of tuples, '
+                    + 'but is a list of something else')
+            jam.annotations.append(f0s_to_jams(f0s))
+
+    # lyrics
+    if lyrics_data is not None:
+        if type(lyrics_data) != list:
+            raise TypeError(
+                'lyrics_data should be a list of tuples')
+        for lyrics in lyrics_data:
+            if type(lyrics) != tuple:
+                raise TypeError(
+                    'lyrics_data should be a list of tuples, '
+                    + 'but is a list of something else')
+            jam.annotations.append(lyrics_to_jams(lyrics))
+
+    return jam
+
+
+def beats_to_jams(beats):
```

Comment:
for readability, i think the function should take two arguments - beat_data and beat_metadata as input.

Reply:
same for the others below

Reply:
mmmm... not sure how to incorporate annotation metadata given the structure it has now, let's chat about it

Reply:
Let's leave it for now and open an issue for after this PR is merged!